src/web3/queryVotingContractEvents.ts

```diff
@@ -0,0 +1,337 @@
+import { ethers } from "ethers";
+import { MAINNET_DEPLOY_BLOCK } from "common/config";
+
+import stringToBytes32 from "common/utils/web3/stringToBytes32";
+
+export interface PriceRound {
+  roundId: string;
+  identifier: string;
+  time: string;
+}
+
+export const queryPriceRoundEvents = async (
+  contract: ethers.Contract,
+  roundId: number | null = null,
+  identifier: string | null = null,
+  time: number | null = null
+) => {
+  // Convert identifier from string to Bytes32.
+  let bytesIdentifier: Uint8Array | null = null;
+  if (identifier) {
+    bytesIdentifier = stringToBytes32(identifier);
+  }
+  // BIG NOTE. You need to pass in null for events with args.
+  // Otherwise this will likely return no values.
+  // PriceRequestAdded(uint256 indexed roundId, bytes32 indexed identifier, uint256 time)
+
+  const priceRequestFilter = contract.filters.PriceRequestAdded(
+    roundId,
+    bytesIdentifier,
+    time
+  );
+
+  try {
+    const events = await contract.queryFilter(
+      priceRequestFilter,
+      MAINNET_DEPLOY_BLOCK
+    );
+
+    return events.map((el) => {
+      const { args } = el;
+      const datum = {} as PriceRound;
+      if (args) {
+        datum.roundId = args[0].toString();
+        datum.identifier = ethers.utils.toUtf8String(args[1]);
+        datum.time = args[2].toString();
+      }
+      return datum;
+    });
+  } catch (err) {
+    console.log("err", err);
+  }
+};
+
+/*  event VoteCommitted(
+  address indexed voter,
+  uint256 indexed roundId,
+  bytes32 indexed identifier,
+  uint256 time,
+  bytes ancillaryData
+);
+*/
+
+export interface VoteEvent {
+  address: string;
+  roundId: string;
+  identifier: string;
+  time: string;
+}
+
+export const queryVotesCommitted = async (
+  contract: ethers.Contract | null,
+  address: string | null = null,
+  roundId: string | null = null,
+  identifier: string | null = null,
+  time: number | null = null
+) => {
+  if (!contract)
```

Comment:
would suggest using assert, its a bit cleaner syntax, though it can be unintuitive, its really `assert.ok`
```js
import assert from 'assert'
assert(contract,"User is not connected to provider and cannot query contract.")
```

Reply:
Yeah that's a nice clean syntax.